config/src/main/java/org/springframework/security/config/annotation/web/builders/WebSecurity.java

```diff
@@ -306,6 +315,28 @@ protected Filter performBuild() throws Exception {
 		return result;
 	}
 
+	private void buildPrivilegeEvaluator(List<SecurityFilterChain> securityFilterChains) {
+		if (this.privilegeEvaluator != null || securityFilterChains.size() <= 1) {
+			return;
+		}
+		List<RequestMatcherDelegatingWebInvocationPrivilegeEvaluator.DelegatePrivilegeEvaluator> delegates = new ArrayList<>();
+		for (SecurityFilterChain securityFilterChain : securityFilterChains) {
+			RequestMatcherDelegatingWebInvocationPrivilegeEvaluator.Builder builder = RequestMatcherDelegatingWebInvocationPrivilegeEvaluator
+					.withRequestMatcher(securityFilterChain::matches);
+			for (Filter filter : securityFilterChain.getFilters()) {
+				if (filter instanceof FilterSecurityInterceptor fsi) {
+					builder.withFilterSecurityInterceptor(fsi, this.servletContext);
+					continue;
+				}
+				if (filter instanceof AuthorizationFilter af) {
+					builder.withAuthorizationManager(af.getAuthorizationManager());
+				}
+			}
+			delegates.add(builder.build());
```

Comment:
Do not add if there is no delegates because we don't want to have to invoke match unnecessarily.